main/src/ca/uwaterloo/flix/runtime/channel/channel.flix

```diff
@@ -0,0 +1,75 @@
+def newId(): ##GlobalCounter & Impure = 
+    import new flix.runtime.GlobalCounter() as newGlobalCounter;
+    import flix.runtime.GlobalCounter:newId() as javaNewId;
+    javaNewId(newGlobalCounter())
+
+type alias Lock = ##java.util.concurrent.locks.Lock
+
+type alias ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
+
+def newReentrantLock(): ReentrantLock & Impure = 
+    import new java.util.concurrent.locks.ReentrantLock() as javaNewReentrantLock;
+    javaNewReentrantLock()
+
+type alias Condition = ##java.util.concurrent.locks.Condition
+
+def newCondition(reentrantLock: ReentrantLock): Condition & Impure =
+    import java.util.concurrent.locks.ReentrantLock.newCondition() as javaNewCondition;
+    javaNewCondition(reentrantLock)
+
+type alias LockConditionPair = {
+    lock: Lock,
+    condition: Condition
+}
+
+type alias Channel[a] = {
+    id: Int32,
+    channelLock: ReentrantLock,
+    bufferSize: Int32,
+    unbuffered: Bool,
+    elementQueue: Array[a],
+    waitingGetters: MutSet[LockConditionPair],
+    waitingSetters: Condition
+}
+
+def newChannel(bufferSize: Int32): Channel & Impure =
+    let unbuffered = default;
```

Comment:
I don't think you want default here. I think you want `0`.

Reply:
What does `default` do? I assumed it just produced the default initialization value of the expected type.

Reply:
> I think you want `0`.

`unbuffered` is supposed to be a boolean, so should we assign it to `false`?

Reply:
`default` does indeed produce a "default" value, but its sort-of a "secret" feature only to be used when absolutely necessary. 

> unbuffered is supposed to be a boolean, so should we assign it to false?

OK